readthedocs/proxito/views/serve.py

```diff
@@ -46,12 +53,20 @@ def get(self, request, subproject_slug=None, filename=""):
 
         unresolved_domain = request.unresolved_domain
         project = unresolved_domain.project
+        parent_project = project
 
         # Use the project from the domain, or use the subproject slug.
         if subproject_slug:
-            project = get_object_or_404(
-                project.subprojects, alias=subproject_slug
-            ).child
+            try:
+                project = project.subprojects.get(alias=subproject_slug).child
+            except ProjectRelationship.DoesNotExist:
+                raise ProxitoSubProjectHttp404(
```

Comment:
This 404 will go directly to the internal nginx redirect (`ServeError404Base`), so it won't work as expected.

Reply:
It replaces a call to `get_object_or_404`, so wouldn't this be exactly the behavior that we want? I.e. replace a general 404 exception with a contextualized one?

Reply:
Yes, but the 404 will go to the internal nginx redirect, it won't hit your custom 404 handler that serves the specific page. And the internal redirect will hit the `ServeError404Base` view, so you won't see this specific exception there.

Reply:
This was much easier to understand after reading the new implementation. It has a lot more clarity than the old implementation. Well done :+1: :+1: